flink-ml-parent/flink-ml-lib/src/test/java/org/apache/flink/ml/pipeline/UserDefinedPipelineStages.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.pipeline;
+
+import org.apache.flink.ml.api.core.Transformer;
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.ml.params.shared.colname.HasSelectedCols;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableEnvironment;
+
+/**
+ * Util class for testing {@link org.apache.flink.ml.api.core.PipelineStage}.
```

Comment:
was wondering why this java class is created here not earlier. is this only to test how pyflink uses JavaTransformer?
Can we reuse the `FakeTransformer` class in TransfomerBaseTest? we can refactor it out or make it public if necessary IMO.

Reply:
Yes, it is used to test `JavaTransformer` including setting/getting params and doing transform(tEnv, table). I find we can't merge these two classes since the transform(tEnv, table) method is different in these two classes. The goal of the two classes is also different. `FakeTransFormer` is used to test transform(BatchOperator) and transform(StreamOperator) while `SelectColumnTransformer` is used to test transform(tEnv, table).  Could we keep the `SelectColumnTransformer` class?

Reply:
+1, seems like there can be some improvement in the Java side testing based on this feedback